java/Makefile

```diff
@@ -0,0 +1,12 @@
+SNAPSHOT_VERSION = $(shell mvn help:evaluate -Dexpression=project.version -q -DforceStdout 2> /dev/null)
+
+version:
+	@echo $(subst -SNAPSHOT,,$(SNAPSHOT_VERSION))
+
+version-set:
+	@mvn versions:set -DnewVersion=$(NEW_VERSION)-SNAPSHOT
+
+release-prepare:
+	@mvn release:prepare
```

Comment:
How is that different to https://www.mojohaus.org/versions-maven-plugin/set-mojo.html which we're using in the `version-set` target?

Reply:
`versions:set` only changes the version number. But a release also includes SCM information.

For example:

```
<scm>
        <connection>scm:git:git://github.com/cucumber/cucumber-jvm.git</connection>
        <developerConnection>scm:git:git@github.com:cucumber/cucumber-jvm.git</developerConnection>
        <url>git://github.com/cucumber/cucumber-jvm.git</url>
        <tag>v7.2.3</tag>
    </scm>
    ```
    
The `tag` is missing `cucumber-expressions` (and here too)` because only `versions:set` was used.

```
    <scm>
        <connection>scm:git:git://github.com/cucumber/cucumber-expressions.git</connection>
        <developerConnection>scm:git:git@github.com:cucumber/cucumber-expressions.git</developerConnection>
        <url>git://github.com/cucumber/cucumber-expressions.git</url>
    </scm>
```

Reply:
But I'm not sure about that  though.

Reply:
Ah. No it doesn't. But:

```
       mvn -Dtag="<tag name>" scm:tag
```

Does.

https://maven.apache.org/scm/maven-scm-plugin/examples/scm-advance-features.html